pytorch_lightning/trainer/optimizers.py

```diff
@@ -205,6 +220,13 @@ def __repr__(self):
         return "No Optimizer"
 
 
+def _validate_optim_conf(optim_conf: Dict[str, Any]) -> None:
+    valid_keys = {"optimizer", "lr_scheduler", "frequency", "monitor"}
+    extra_keys = [k for k in optim_conf.keys() if k not in valid_keys]
+    if extra_keys:
+        rank_zero_warn(f"Found unsupported keys in the optimizer configuration: {extra_keys}", RuntimeWarning)
```

Comment:
should it raise an exception here? I have seen users passing scheduler config like `interval` and `scheduler` in optimizer config and then expects it to work.. warnings are ignored sometimes but atleast it should stop at the beginning rather than releasing the mistake after an epoch/some epochs while logging at the logs.

Reply:
I don't think so considering they will just get ignored and shouldn't impact the flow.

Reply:
Exactly my point.. they will be ignored with a warning and it will continue to run with possibly due to the incorrect config.. and possibly user might not know about this if they ignore the warning. Since optimizers and schedulers highly impact the training, we should ensure they are correct from users side and tell them at the beginning only.

Reply:
we could consider typing the dict for users who want stricter checking with mypy. something for the future maybe.

Reply:
> I don't think so considering they will just get ignored and shouldn't impact the flow.

is it a misconfiguration from the user POV? as in, they input these keys expecting them to take effect but they don't? If so, i'd be in favor of raising an exception to warn earlier. 

i am also in favor of a typed dict/dataclass that users can return which makes the schema explicit and makes it harder to input bad configs

Reply:
Please let me know how I should proceed with this PR. Should I raise an exception and create a typed dict?

Reply:
I'd say create a `TypedDict`.

This has the advantage of easily seeing all fields and expected types. With an exception you have to convey the information through text which is not as useful.

Additionally, raising an exception can be a breaking change in case people use `configure_optimizers` for other things.

The `TypedDict` improvement should be done in a follow-up PR